visvis2/renderers/_wgpu.py

```diff
@@ -33,18 +35,37 @@ def __init__(self, canvas):
             wgpu.TextureUsage.OUTPUT_ATTACHMENT,
         )
 
-    def traverse(self, obj):
-        yield obj
-        for child in obj.children:
-            yield from self.traverse(child)
+    def get_render_list(self, scene: WorldObject, proj_screen_matrix: Matrix4):
+        # start by gathering everything that is visible and has a material
+        q = []
+
+        def visit(wobject):
+            nonlocal q
+            if wobject.visible and hasattr(wobject, "material"):
+                q.append(wobject)
+
+        scene.traverse(visit)
+
+        # next, sort them from back-to-front
+        def sort_func(wobject: WorldObject):
+            z = (
+                Vector3()
+                .set_from_matrix_position(wobject.matrix_world)
+                .apply_matrix4(proj_screen_matrix)
+                .z
+            )
+            return wobject.render_order, z
```

Comment:
`z` may need a `-` but I can't really tell until we test more complex scenes. it should prevent some overdraw :)